src/main/java/net/imagej/ops/segment/hough/HoughTransformOpNoWeights.java

```diff
@@ -0,0 +1,116 @@
+package net.imagej.ops.segment.hough;
+
+import org.scijava.ItemIO;
+import org.scijava.Priority;
+import org.scijava.app.StatusService;
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+
+import net.imagej.ops.Contingent;
+import net.imagej.ops.special.hybrid.AbstractUnaryHybridCF;
+import net.imglib2.Cursor;
+import net.imglib2.Dimensions;
+import net.imglib2.FinalDimensions;
+import net.imglib2.IterableInterval;
+import net.imglib2.img.Img;
+import net.imglib2.img.ImgFactory;
+import net.imglib2.type.BooleanType;
+import net.imglib2.type.numeric.real.DoubleType;
+import net.imglib2.view.IntervalView;
+import net.imglib2.view.Views;
+
+/**
+ * Hough transform for binary images.
+ * 
+ * @author Jean-Yves Tinevez
+ *
+ * @param <T>
+ *            the type of source image. Must extend boolean type.
+ */
+@Plugin( type = HoughCircleTransformOp.class, priority = Priority.HIGH )
+public class HoughTransformOpNoWeights< T extends BooleanType< T > >
+		extends AbstractUnaryHybridCF< IterableInterval< T >, Img< DoubleType > >
+		implements Contingent
+{
+
+	@Parameter
+	protected StatusService statusService;
+
+	@Parameter( label = "Min circle radius", description = "Minimal radius, in pixel units, for the transform.", min = "1", type = ItemIO.INPUT )
+	protected long minRadius = 1;
+
+	@Parameter( label = "Max circle radius", description = "Maximal radius, in pixel units, for the transform.", min = "1", type = ItemIO.INPUT )
+	protected long maxRadius = 50;
+
+	@Parameter( label = "Step radius", description = "Radius step, in pixel units, for the transform.", min = "1", type = ItemIO.INPUT, required = false )
+	protected long stepRadius = 1;
+
+
+	@Override
+	public Img< DoubleType > createOutput( final IterableInterval< T > input )
+	{
+		// Get a suitable image factory.
+		final int numDimensions = input.numDimensions();
+		if ( input.numDimensions() != 2 ) { throw new IllegalArgumentException(
+				"Cannot compute Hough circle transform for non-2D images. Got " + numDimensions + "D image." ); }
```

Comment:
Ok.

Reply:
So I think that the best solution to this problem is to actually keep the Contingent interface and remove the runtime check. @ctrueden and I discussed a few reasons for this.
*The Contingent interface is used to prevent Op Matching based on factors other than input types, e.g. Number of Dimensions, like you had before. If we keep the Contingent then it will simply overlook the op if we input a non-2D image, instead of attempting the op and throwing an error.
*Let's say that, in the future, we want to make a Hough Circle/Sphere Transform that takes in only 3 Dimensions, or something else along these lines that does not involve the op signature itself. We would have to refactor this op instead of make a new one, which goes against the extensible case-logic that Ops should be.
*The run-time checking could potentially be time consuming. Let's say that instead of using `ops().run()` the user calls `ops().op()`, which would return an Op reference that the user could use to run through a bunch of images, instead of calling `ops().run()` with every image. While the user could theoretically pass an image into this reference that would not be caught by `conforms`, it is part of the social contract with the user that they will not do this. And if there are a lot of images run over this reference, then it could hamper run time if the reference is used many times.

Reply:
Ok!